server/controllers/finance/reports/balance_sheet/index.js

```diff
@@ -1,240 +1,153 @@
-/**
- * Balance sheet Controller
- *
- * This controller is responsible for processing
- * the balance sheet (bilan) report.
- *
- * @module reports/balance_sheet
- *
- * @requires lodash
- * @requires lib/db
- * @requires lib/ReportManager
- * @requires lib/errors/BadRequest
- */
-
+const q = require('q');
 const _ = require('lodash');
-const moment = require('moment');
 const db = require('../../../../lib/db');
+const util = require('../../../../lib/util');
+const Tree = require('../../../../lib/Tree');
 const ReportManager = require('../../../../lib/ReportManager');
 
-// report template
-const TEMPLATE = './server/controllers/finance/reports/balance_sheet/report.handlebars';
+const fiscal = require('../../fiscal');
 
-const ASSET = 1;
-const LIABILITY = 2;
-const EQUITY = 3;
-const INCOME = 4;
-const EXPENSE = 5;
-const DATE_FORMAT = 'YYYY-MM-DD';
-const FC_CURRENCY = 1;
+const TEMPLATE = './server/controllers/finance/reports/balance_sheet/report.handlebars';
 
-// expose to the API
 exports.document = document;
+exports.formatData = formatData;
+
+
+const ASSET_ACCOUNT_TYPE = 1;
+const LIABILITY_ACCOUNT_TYPE = 2;
+const EQUITY_ACCOUNT_TYPE = 3;
+
+const DECIMAL_PRECISION = 2; // ex: 12.4567 => 12.46
 
-/**
- * @function document
- * @description process and render the balance report document
- */
 function document(req, res, next) {
   const params = req.query;
-  const session = {};
-  const bundle = {};
-  let report;
-
-  // date options
-  if (params.dateFrom && params.dateTo) {
-    session.dateFrom = moment(params.dateFrom).format(DATE_FORMAT);
-    session.dateTo = moment(params.dateTo).format(DATE_FORMAT);
-  } else {
-    session.date = moment(params.date).format(DATE_FORMAT);
-  }
-
-  session.enterprise = req.session.enterprise;
-  params.enterpriseId = session.enterprise.id;
 
-  _.defaults(params, { user : req.session.user });
+  let docReport;
+  const options = _.extend(req.query, {
+    filename : 'TREE.OPERATING_ACCOUNT',
+    csvKey : 'rows',
+    user : req.session.user,
+  });
 
   try {
-    report = new ReportManager(TEMPLATE, req.session, params);
+    docReport = new ReportManager(TEMPLATE, req.session, options);
```

Comment:
A curiosity question, is there any reason to rename this variable to `docReport` ?